### ISSUE TITLE
WebjarService broken on Windows for assets with more than 3 path components

### ISSUE THREAD

**Issue by fthomas** [http4s/http4s#3304]
I think [this change](https://github.com/http4s/http4s/commit/464060da6c146e3b96a65843f62168e9658c4eb1#diff-be6c03bb38414d6fc44166edc99f910aR98) (which was part of the fix for GHSA-66q9-f7ff-mmx6) has partially broken the `WebjarService` on Windows. A `Path` with more than three components will result in a `WebjarAsset` where the `asset` field contains backslashes as path separator. The `asset` field is then later used to load a resource which fails because the resource name contains backslashes. For example, trying to load the webjar resource `bootstrap/3.3.7/js/bootstrap.js` results in a `WebjarAsset("bootstrap", "3.3.7", "js\bootstrap.js")` which is then later used to load the resource `"META-INF/resources/webjars/bootstrap/3.3.7/js\bootstrap.js"`.

I guess a simple fix could be to use
```scala
import scala.jdk.CollectionConverters._
asset.iterator().asScala.mkString("/")
```
instead of `asset.toString` so we're not relying on the OS-specific separators.

**Comment by rossabaker** [comment]
I don't have a way to test this.  Do you want to try a PR, or would you be willing to retest a fix?  Ideally, we'd target series/0.20.  (I don't know that we want to dig up 0.18 yet again for this, since it's not a security flaw, per se.)

**Comment by fthomas** [comment]
I just confirmed that the fix above works for me. I'll prepare a PR.
